edk2toollib/uefi/edk2/path_utilities.py

```diff
@@ -161,9 +160,12 @@ def GetContainingPackage(self, InputPath):
                 return None
 
         # InputPath is in workspace or PackagesPath for worst case scenario.
-
-        dirpathprevious = os.path.dirname(InputPath)
-        dirpath = os.path.dirname(InputPath)
+        if os.path.isdir(InputPath):
```

Comment:
do we have unit test coverage on both of these code paths?  I assume not given that this bug didn't get caught until integration testing.

Reply:
I've added a test case to cover the additional codepath of if it is a folder instead of a file.